firmware/common/config/dispatch.config

```diff
@@ -10,343 +10,765 @@
-
-	function 4										// Function 4 is open file
-		*DERROR = FIOOpenFileHandle(DPARAMS[0],DSPGetStdString(DPARAMS,1),DPARAMS[3]);
-		
-	function 5										// Function 5 is close file
-		*DERROR = FIOCloseFileHandle(DPARAMS[0]);
-
-	function 6										// Function 6 is seek to position
-		*DERROR = FIOSeekFileHandle(DPARAMS[0],DSPGetInt32(DPARAMS,1));
-
-	function 7										// Function 7 is tell position
-	{
-		uint32_t pos;
-		*DERROR = FIOTellFileHandle(DPARAMS[0],&pos);
-		DSPSetInt32(DPARAMS,1,pos);
-	}
-
-	function 8                                      // Function 8 is read data
-	{
-		uint16_t size = DSPGetInt16(DPARAMS,3);
-		*DERROR = FIOReadFileHandle(DPARAMS[0],DSPGetInt16(DPARAMS,1),&size);
-		DSPSetInt16(DPARAMS,3,size);
-	}
-
-	function 9                                      // Function 9 is write data
-	{
-		uint16_t size = DSPGetInt16(DPARAMS,3);
-		*DERROR = FIOWriteFileHandle(DPARAMS[0],DSPGetInt16(DPARAMS,1),&size);
-		DSPSetInt16(DPARAMS,3,size);
-	}
-
-	function 10                                     // Function 10 is get size
-	{
-		uint32_t size;
-		*DERROR = FIOGetSizeFileHandle(DPARAMS[0],&size);
-		DSPSetInt32(DPARAMS,1,size);
-	}
-
-	function 11                                     // Function 11 is set size
-	{
-		*DERROR = FIOSetSizeFileHandle(DPARAMS[0],DSPGetInt32(DPARAMS,1));
-	}
-
-	function 12										// Function 12 is rename
-	{
-		*DERROR = FIORenameFile(DSPGetStdString(DPARAMS, 0), DSPGetStdString(DPARAMS, 2));
-	}
-	
-	function 13										// Function 13 is delete
-	{
-		*DERROR = FIODeleteFile(DSPGetStdString(DPARAMS, 0));
-	}
-	
-	function 14										// Function 14 is create directory
-	{
-		*DERROR = FIOCreateDirectory(DSPGetStdString(DPARAMS, 0));
-	}
-	
-	function 15										// Function 15 is change directory
-	{
-		*DERROR = FIOChangeDirectory(DSPGetStdString(DPARAMS, 0));
-	}
-	
-	function 16										// Function 16 is stat file
-	{
-		uint32_t length;
-		uint8_t type;
-		*DERROR = FIOStatFile(DSPGetStdString(DPARAMS, 0), &length, &type);
-		DSPSetInt32(DPARAMS, 0, length);
-		DPARAMS[4] = type;
-	}
-
-	function 17                                     // Function 17 is open directory
-	{
-		*DERROR = FIOOpenDir(DSPGetStdString(DPARAMS, 0));
-	}
-
-	function 18                                     // Function 18 is read directory
-	{
-		std::string filename;
-		uint32_t size;
-		uint8_t attribs;
-		*DERROR = FIOReadDir(filename, &size, &attribs);
-		DSPSetStdString(DPARAMS, 0, filename);
-		DSPSetInt32(DPARAMS, 2, size);
-		DPARAMS[6] = attribs;
-	}
-	
-	function 19										// Function 19 is close directory
-	{
-		*DERROR = FIOCloseDir();
-	}
-
-	function 20										// Function 20 copies a file
-		*DERROR = FIOCopyFile(DSPGetStdString(DPARAMS, 0), DSPGetStdString(DPARAMS, 2));
-
-	function 32 									// Function 32 is directory listing with wildcard
-		FIODirectory(DSPGetString(DCOMMAND,4));
+GROUP 3 ,0,9 File I/O
+// Q3: 'File I/O' specifically, or someday generic 'I/O' (eg: physical pins)?
```

Comment:
this one answered itself "yes, 'File I/O' specifically" - group 10 is generic I/O